bundles/paikkatietoikkuna/mapmodule/mapmodule.olcs.js

```diff
@@ -454,6 +456,122 @@ class MapModuleOlCesium extends MapModuleOl {
     }
 
+    /**
+     *
+     * @method centerMap
+     * Moves the map to the given position and zoomlevel. Overrides 2d centerMap function.
+     * @param {Number[] | Object} lonlat coordinates to move the map to
+     * @param {Number/OpenLayers.Bounds/Object} zoomLevel zoomlevel to set the map to
+     * @param {Boolean} suppressEnd true to NOT send an event about the map move
+     *  (other components wont know that the map has moved, only use when chaining moves and
+     *     wanting to notify at end of the chain for performance reasons or similar) (optional)
+     * @param {Object} options  has values for heading, pitch, roll and duration
+     */
+    centerMap (lonlat, zoom, suppressEnd, options) {
+        lonlat = this.normalizeLonLat(lonlat);
+        const location = olProj.transform([lonlat.lon, lonlat.lat], this.getProjection(), 'EPSG:4326');
+        const cameraHeight = this.adjustZoom(zoom);
+        const duration = options && options.duration ? options.duration : 3000;
+        const animationDuration = duration / 1000;
+        const camera = options && options.heading && options.roll && options.pitch
+            ? { heading: options.heading,
+                roll: options.roll,
+                pitch: options.pitch } : undefined;
+
+        if (options && options.animation) {
+            this._flyTo(location[0], location[1], cameraHeight, animationDuration, camera);
+            if (suppressEnd !== true) {
+                setTimeout(() => {
+                    this.notifyMoveEnd();
+                }, duration);
+            }
+            return true;
+        } else {
+            this.getMap().getView().setCenter([lonlat.lon, lonlat.lat]);
+            this.getMap().getView().setZoom(zoom.value);
+            this.notifyMoveEnd();
+        }
+    }
+
+    /**
+     * @method _flyTo fly to coords with options
+     *
+     * @param {Number} x longitude
+     * @param {Number} y latitude
+     * @param {Number} z zoom/cameraheight
+     * @param {Number} duration animation duration in seconds
+     * @param {Object} camera orientation of camera, heading pitch and roll
+     * @param {Function} complete function to run when animation is completed
+     * @param {Function} cancel function to run when animation is cancelled
+     */
+    _flyTo (x, y, z, duration, cameraAngles, complete, cancel) {
+        const camera = this.getCesiumScene().camera;
+        let flyToParams = { destination: Cesium.Cartesian3.fromDegrees(x, y, z) };
+        flyToParams = duration ? { ...flyToParams, duration: duration } : flyToParams;
+        flyToParams = cameraAngles ? { ...flyToParams, orientation: cameraAngles } : flyToParams;
+        flyToParams = complete ? { ...flyToParams, complete: complete } : flyToParams;
+        flyToParams = cancel ? { ...flyToParams, cancel: cancel } : flyToParams;
+        camera.flyTo(flyToParams);
+    }
+
+    adjustZoom (zoom) {
+        if (zoom === null || zoom === undefined) {
```

Comment:
Should this handle the zoomToExtent?

Reply:
I will handle it on centerMap for now, should probably finetune those values and/or logic on this.